xdsl/dialects/builtin.py

```diff
@@ -554,6 +558,9 @@ def from_params(
                                    bound=Attribute,
                                    covariant=True)
 
+_UnrankedTensorTypeInitElems = TypeVar("_UnrankedTensorTypeInitElems",
```

Comment:
Very general question, not super important for this PR: Do we want separate type vars for each attribute, even when they are the same? I don't really see the point.

Reply:
Agreed!

Reply:
That's my fault I believe, I always wanted to duplicate them. But I'm fine saying that we should not do that!

Also, before I was against using them across files, since they were creating silent errors (when we removed one and forgot about its uses). But since we are now using pyright in the CI, I'm fine reusing them, and probably we should put them in the `hints` file for the most common one?

(In another PR, of course)

Reply:
I'll file an issue, would be good to clean up indeed.

Reply:
#515